docker/docker_client.go

```diff
@@ -48,6 +48,15 @@ const (
 	extensionSignatureTypeAtomic    = "atomic" // extensionSignature.Type
 )
 
+// systemPerHostCertDirPath is the path to certs.d, used for storing per-host TLS keys and certificates.
+// You can override this at build time with
+// -ldflags '-X github.com/containers/image/docker.systemPerHostCertDirPath=$your_path'
+var systemPerHostCertDirPath = builtinPerHostCertDirPath
```

Comment:
I don't understand why are we making this configurable at compile time. This is the docker transport and as such, the default is `/etc/docker/certs.d` unless I'm missing something. Configuring this at compile time goes against what we already have for registries auths with `$HOME/.docker/config.json` - we don't let ppl override that at compile time. That's something implementation specific and I believe `/etc/docker/certs.d` is as well.

I feel I'm missing something as to why we need this compile time stuff so bear with me.